src/typing/type_inference_js.ml

```diff
@@ -1135,7 +1133,14 @@ and statement cx = Ast.Statement.(
   | (loc, Empty) -> ()
 
   | (loc, Block { Block.body }) ->
-      toplevels cx body
+      let entries = ref SMap.empty in
+      let block = { scope = Block; entries } in
+      Env_js.push_env block;
+
+      List.iter (statement_decl cx) body;
```

Comment:
I think this is ok -- it's essentially what functions currently do (except way over in the `mk_body` function)